cmake/SetupGsl.cmake

```diff
@@ -8,7 +8,7 @@ list(APPEND SPECTRE_LIBRARIES ${GSL_LIBRARIES})
 # Extract the path where the shared libraries are and point the linker to
 # that directory
 list(GET GSL_LIBRARIES 0 FIRST_GSL_LIBRARY)
-string(REGEX REPLACE "libgsl[cblas]*.so" ""
+string(REGEX REPLACE "libgsl[cblas]*.*" ""
```

Comment:
I think the regex we want is: `libgsl(cblas)?.(so|a|dylib)`. This will match fewer things, but also be more targeted. I'm not sure if there are ever situations where the gsl lib so is `.so.SOME_NUMBER`. The macOS case should be handled by the `dylib`